src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -442,22 +451,96 @@ public static GapicProductConfig createDummyInstance(
       protoInterfaces.put(interfaceConfigProto.getName(), apiInterface);
     }
 
-    // Generate GapicInterfaceConfigs from the proto interfaces and interfaceConfigProtos.
+    // Store info about each Interface in a GapicInterfaceInput object.
+    ImmutableList.Builder<GapicInterfaceInput> interfaceInputs = ImmutableList.builder();
     for (Entry<String, Interface> interfaceEntry : protoInterfaces.entrySet()) {
       String serviceFullName = interfaceEntry.getKey();
       InterfaceConfigProto interfaceConfigProto =
           interfaceConfigProtos.getOrDefault(
               serviceFullName, InterfaceConfigProto.getDefaultInstance());
       Interface apiInterface = interfaceEntry.getValue();
+
+      GapicInterfaceInput interfaceInput =
+          createInterfaceInput(
+              apiInterface, interfaceConfigProto, diagCollector, gapicConfigPresent);
+      if (interfaceInput == null) {
+        return null;
+      }
+      interfaceInputs.add(interfaceInput);
+    }
+
+    return interfaceInputs.build();
+  }
+
+  /** Return a data object containing the client entities to be generated. */
+  private static GapicInterfaceInput createInterfaceInput(
```

Comment:
Rather than have one method with a bool param, can this be two methods? There is arguably not really any shared code between the two cases, except for creating the builders.

Reply:
hokay